tools-local/tasks/BuildFPMToolPreReqs.Netstandard2_0.cs

```diff
@@ -0,0 +1,581 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// This task prepares the command line parameters for running a RPM build using FPM tool and also updates the copyright and changelog file tokens.
+    /// If parses various values from the config json by first reading it into a model and then builds the required string for parameters and passes it back.
+    /// 
+    /// </summary>
+    public class BuildFPMToolPreReqs : BuildTask
+    {
+        [Required]
+        public string InputDir { get; set; }
+        [Required]
+        public string OutputDir { get; set; }
+        [Required]
+        public string PackageVersion { get; set; }
+        [Required]
+        public string ConfigJsonFile { get; set; }
+        [Output]
+        public string FPMParameters { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                if (!File.Exists(ConfigJsonFile))
+                {
+                    throw new FileNotFoundException($"Expected file {ConfigJsonFile} was not found.");
+                }
```

Comment:
Won't File.OpenText throw the same exception, and without the inherent race condition?  You're just trying to provide a better exception message in some cases?

Reply:
Originally written this way: https://github.com/dotnet/core-setup/pull/2462/files#diff-0ebce9e49dc50f19e82e0d3ceada8568R43